javatests/arcs/core/allocator/AllocatorTestBase.kt

```diff
@@ -173,6 +203,213 @@ open class AllocatorTestBase {
+    val particle = Plan.Particle(
+      particleName = "Unknown Particle",
+      location = "unknown.Particle",
+      handles = emptyMap()
+    )
+
+    invariant_addUnmappedParticle_generatesError(
+      Value(PersonPlan),
+      Value(hostRegistry),
+      Value(particle)
+    )
+  }
+
+  /**
+   * Generate a random handleConnection map.
+   */
+  fun handleMapGenerator(s: Seed): A<Map<String, Plan.HandleConnection>> {
+    val storageKey = ChooseFromList(s, listOf("sk1", "sk2", "sk3", "sk4", "sk5", "sk6"))
+    return MapOf(
+      SequenceOf(listOf("a", "b", "c", "d", "e")),
+      HandleConnectionGenerator(
+        handle = HandleGenerator(
+          storageKey = CreatableStorageKeyGenerator(storageKey),
+          // TODO(shanestephens): Add a type generator
+          type = Value(PersonPlan.handles[0].type)
+        ),
+        mode = HandleModeFromType(s),
+        type = Value(PersonPlan.handles[0].type)
+      ),
+      ChooseFromList(s, listOf(1, 2, 3, 4, 5))
+    )
+  }
+
+  /**
+   * Test that adding a randomly generated unmapped particle will result in a plan being unable to
+   * be started.
+   */
+  @Test
+  open fun fuzz_addUnmappedParticle_generatesError() = runFuzzTest {
+    val particle = PlanParticleGenerator(
+      ChooseFromList(it, listOf("Particle", "Shmarticle", "Blarticle", "Fred", "Jordan", "Bob")),
+      ChooseFromList(it, listOf("location.one", "a.location.two", "the.location.three")),
+      handleMapGenerator(it)
+    )
+
+    invariant_addUnmappedParticle_generatesError(Value(PersonPlan), Value(hostRegistry), particle)
+  }
+
+  /**
+   * Test that PersonPlan can be started with a hostRegistry established for this purpose.
+   */
+  @Test
+  open fun allocator_canPartitionArcInExternalHosts() = runAllocatorTest {
+    val particleInfos = listOf(
+      ParticleInfo(::WritePerson.toRegistration(), writePersonParticle),
+      ParticleInfo(::ReadPerson.toRegistration(), readPersonParticle),
+      ParticleInfo(::PurePerson.toRegistration(), purePersonParticle)
+    )
+
+    invariant_planWithOnly_mappedParticles_willResolve(
+      Value(particleInfos),
+      Function { PersonPlan },
+      Function { hostRegistry }
+    )
+  }
+
+  /**
+   * Test that a randomly generated plan will resolve against a randomly generated registry,
+   * as long as the registry hosts all particles in the plan.
+   */
+  @Test
+  open fun fuzz_planWithOnly_mappedParticles_willResolve() = runFuzzTest {
+    val names = ChooseFromList(
+      it,
+      listOf("Particle", "Shmarticle", "Blarticle", "Fred", "Jordan", "Bob")
+    )
+    val particles = ListOf(ParticleInfoGenerator(it, names, handleMapGenerator(it)), Value(10))
+    val planT = PlanFromParticles(it)
+
+    val registryT = HostRegistryFromParticles(it)
+
+    invariant_planWithOnly_mappedParticles_willResolve(particles, planT, registryT)
+  }
+
+  /**
+   * REGRESSION for planWithOnly_mappedParticles_willResolve.
```

Comment:
does it mean fuzz_planWithOnly_mappedParticles_willResolve will fail sometimes? in that case, do we need to disable it?

Reply:
No, it's demonstrating how to create a regression test.